libcxx/test/std/experimental/simd/test_util.h

```diff
@@ -13,32 +13,37 @@
 
 namespace ex = std::experimental::parallelism_v2;
 
-template <class F>
-void test_simd() {}
+template <class F, class _Tp, class Simd, class... Simds>
+void test_simd_abi();
 
-template <class F, class Simd, class... Simds>
-void test_simd() {
-  F{}.template operator()<Simd>();
-  test_simd<F, Simds...>();
+template <class F, class _Tp>
+void test_all_simd_abi() {
+  using namespace ex;
+  // TODO: generate number sequence to test different length `simd abi`
+  test_simd_abi<F, _Tp, simd_abi::native<_Tp>, simd_abi::fixed_size<42>, simd_abi::__scalar, simd_abi::compatible<_Tp>,
```

Comment:
`simd_abi::__scalar` is an internal interfaces that will not be exposed to external users. It should be `simd_abi::scalar` here.

The c++ standard library retains the prefix formats of double underlined with lowercase letters and single underlined with uppercase letters as the definitions required for internal implementation. Do not use any internal interfaces in the test.